### PR TITLE
Feature/spm cli #6

### PR DESCRIPTION
**Pull request by panicmilos**

### CODE REVIEW

File: lang/src/spm-cli/cli.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  pass
+
+cli.add_command(init)
+cli.add_command(install)
+cli.add_command(deploy)
+cli.add_command(version)
+cli.add_command(pack)
+
+if __name__ == '__main__':
+  cli()
```
Comment (bjelicaluka) [inline lang/src/spm-cli/cli.py:19]: this looks awesome!

File: lang/src/spm-cli/constants.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+PACKAGE_JSON_PATH = "package.json"
+SPM_PACKAGES_PATH = "spm_packages"
+
+
+BASE_URL = "http://host.docker.internal" # "http://localhost"
+SEARCH_INDEX_API_URL = f"{BASE_URL}:3000/api"
```
Comment (bjelicaluka) [inline lang/src/spm-cli/constants.py]: change to: `https://spm.bjelicaluka.com`

File: lang/src/spm-cli/constants.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+IPFS_NODE_URL = f"{BASE_URL}:9090/ipfs"
```
Comment (bjelicaluka) [inline lang/src/spm-cli/constants.py]: change to: `https://gw.spm.bjelicaluka.com`

File: lang/src/spm-cli/crypto.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  return priv_key.to_string().hex(), pub_key.to_string().hex()
+
+
+def create_signature(priv_key_string, text):
+  priv_key = ecdsa.SigningKey.from_string(
+      bytes.fromhex(priv_key_string),
+      curve = ecdsa.SECP256k1,
+      hashfunc = sha256
+
```
Comment (bjelicaluka) [inline lang/src/spm-cli/crypto.py]:
```suggestion
```

File: lang/src/spm-cli/init.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  with open(PACKAGE_JSON_PATH, "w") as fp:
+    fp.write(json.dumps(init_content, sort_keys=True, indent=4))
+
+
+def read_package_data():
+  name = take_input("Package name")
+  author = take_input("Package author")
+  
+  while True:
+    version = take_input("Package version", "1.0.0")
```
Comment (bjelicaluka) [inline lang/src/spm-cli/init.py]: nit: Does it make sense to show the default version? If it's used only for package init, then it does make sense, but I would change it to "0.0.1".

File: lang/src/spm-cli/init.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    if is_version_valid(version):
+      break
+    print("Version must be Major.Minor.Patch")
```
Comment (bjelicaluka) [inline lang/src/spm-cli/init.py]:
```suggestion
    print("The prefered version format is: Major.Minor.Patch, where each part is a whole number.")
```

File: lang/src/spm-cli/install.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  if version == '':
+    version = "latest"
+
+  install_single_package(name, version)
+
+if __name__ == "__main__":
+  install()
+  # install_single_package("test_package", "latest")
+  # install_single_package("test_package2", "1.0.0")
+  # install_from_package_json()
```
Comment (bjelicaluka) [inline lang/src/spm-cli/install.py:125]: either remove or add a NOTE that it's used for manual testing

**Review by bjelicaluka — APPROVED** [review]
Awesome! The code is super clean and easy to read. I love the way it is structured.
I have a few nits, but I'll approve. Please just make sure to change the URLs before merging.

Also, it would be nice to provide some information on how to install the CLI, and maybe think about how we're going to deploy it.

File: lang/src/spm-cli/init.py (hunk 6 above)
Comment (panicmilos) [reply lang/src/spm-cli/init.py]: What a story writer 😄

**Pull request merged**